lib/src/list/list_builder.dart

```diff
@@ -227,15 +232,20 @@ class ListBuilder<E> {
 
   // Internal.
 
-  ListBuilder._fromBuiltList(BuiltList<E> builtList)
-      : _copyBeforeWrite = true,
-        _builtList = builtList,
-        _list = builtList._list {
+  ListBuilder._uninitialized() {
     _checkGenericTypeParameter();
   }
 
-  ListBuilder._withSafeList(this._list) : _copyBeforeWrite = false {
-    _checkGenericTypeParameter();
+  void _replaceFromBuiltList(BuiltList<E> builtList) {
+    _copyBeforeWrite = true;
+    _builtList = builtList;
+    _list = builtList._list;
+  }
+
+  void _replaceFromSafeList(List<E> list) {
```

Comment:
set _safeList ?

And take an Iterable and call .toList(growable: true) on it and call it from all places where you assign to _list?